client/src/app/routes/stardust/OutputsList.tsx

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { IOutputResponse } from "@iota/iota.js-stardust";
+import React, { useEffect, useState } from "react";
+import { RouteComponentProps, useLocation } from "react-router-dom";
+import { ServiceFactory } from "../../../factories/serviceFactory";
+import { STARDUST } from "../../../models/config/protocolVersion";
+import { StardustTangleCacheService } from "../../../services/stardust/stardustTangleCacheService";
+import Pagination from "../../components/Pagination";
+import Spinner from "../../components/Spinner";
+import Output from "../../components/stardust/Output";
+import "./OutputsList.scss";
+import OutputsListProps from "./OutputsListProps";
+
+interface OutputsListState {
+    outputIds: string[];
+    tag: string;
+}
+
+interface OutputsListResponse {
+    output: IOutputResponse;
+    outputId: string;
+}
+
+const TOKEN_PAGE_SIZE: number = 5;
+
+const OutputsList: React.FC<RouteComponentProps<OutputsListProps>> = (
+    { match: { params: { network } } }
+) => {
+    const [outputsDetail, setOutputsDetail] = useState<OutputsListResponse[] | undefined>();
+    const { state } = useLocation<OutputsListState>();
+    const [currentPage, setCurrentPage] = useState<OutputsListResponse[]>([]);
+    const [pageNumber, setPageNumber] = useState(1);
+    const [isLoading, setIsLoading] = useState(true);
+    const { outputIds, tag } = state ?? {
+        outputIds: [],
+        tag: ""
+    };
+
+    useEffect(() => {
```

Comment:
Probably here you can just call `getOutputDetails();`, and that's it (you don't need to pass the parameter, as it will be available inside the function globally), and then handle everything in the function (case where no `outputIds`, and `setLoading(false`)